lib/svg_sprite/svg_sprite.rb

```diff
@@ -415,28 +421,34 @@ def self.prepare_symbol(symbol, svg_filename)
   end
 
   def self.settings_icons
-    # includes svg_icon_subset and any settings containing _icon (incl. plugin settings)
-    site_setting_icons = []
+    get_set_cache("settings_icons") do
+      # includes svg_icon_subset and any settings containing _icon (incl. plugin settings)
+      site_setting_icons = []
 
-    SiteSetting.settings_hash.select do |key, value|
-      if key.to_s.include?("_icon") && String === value
-        site_setting_icons |= value.split('|')
+      SiteSetting.settings_hash.select do |key, value|
+        if key.to_s.include?("_icon") && String === value
+          site_setting_icons |= value.split('|')
+        end
       end
-    end
 
-    site_setting_icons
+      site_setting_icons
+    end
   end
 
   def self.plugin_icons
     DiscoursePluginRegistry.svg_icons
   end
 
   def self.badge_icons
-    Badge.pluck(:icon).uniq
+    get_set_cache("badge_icons") do
```

Comment:
The `Badge` model expires this cache each time a badge is updated.

Reply:
https://github.com/discourse/discourse/blob/eb2c399445bf1007c9eb3c358003191ffcec0211/app/models/badge.rb#L121-L122